### PR TITLE
@damassi => Add new set of commercial filtering options on artist page filter

### PR DESCRIPTION
**Pull request by mzikherman**
controlled by lab feature

For storybook purposes, you can set `USER_LAB_FEATURES=Lab1,Lab2,Lab3` a comma separate string of lab features to turn on (since we're not fetching current user).

When coming from Force, `req.user` will have a `lab_features` array.

### CODE REVIEW

File: .storybook/webpack.config.js [hunk 1]
```diff
@@ -52,6 +53,7 @@ if (USER_ID && USER_ACCESS_TOKEN) {
       "process.env": {
         USER_ID: JSON.stringify(USER_ID),
         USER_ACCESS_TOKEN: JSON.stringify(USER_ACCESS_TOKEN),
+        USER_LAB_FEATURES: JSON.stringify(USER_LAB_FEATURES),
```
Comment (mzikherman) [inline .storybook/webpack.config.js]: injects the ENV var into the special `process.env` sandbox that storybooks uses.

File: src/Apps/Artist/Routes/Overview/Components/ArtworkFilter/ArtworkFilterRefetch.tsx [hunk 2]
```diff
@@ -81,6 +81,8 @@ export const ArtworkFilterRefetchContainer = createRefetchContainer(
           major_periods: { type: "[String]" }
           partner_id: { type: "ID" }
           for_sale: { type: "Boolean" }
+          at_auction: { type: "Boolean" }
```
Comment (mzikherman) [inline src/Apps/Artist/Routes/Overview/Components/ArtworkFilter/ArtworkFilterRefetch.tsx]: this just propagates the query vars around, from the query renderer -> fragments.

File: src/Apps/Artist/Routes/Overview/Components/ArtworkFilter/index.tsx [hunk 3]
```diff
@@ -81,19 +138,17 @@ class Filter extends Component<Props> {
...
-                              selected={filters.state.for_sale}
-                              disabled={!hasForSaleArtworks}
-                              onSelect={value => {
-                                return filters.setFilter(
-                                  "for_sale",
-                                  value,
+                            {currentUser &&
+                            currentUser.lab_features &&
+                            currentUser.lab_features.indexOf(
+                              "New Buy Now Flow"
```
Comment (mzikherman) [inline src/Apps/Artist/Routes/Overview/Components/ArtworkFilter/index.tsx]: piggy backing on this lab feature

File: src/Apps/__stories__/Apps.story.tsx [hunk 4]
```diff
@@ -17,7 +17,7 @@ storiesOf("Apps", module)
     return (
       <StorybooksRouter
         routes={artistRoutes}
-        initialRoute="/artist/pablo-picasso"
+        initialRoute="/artist/walter-gropius"
```
Comment (mzikherman) [inline src/Apps/__stories__/Apps.story.tsx]: i can undo, this artist has an ecommerce work available on staging

File: src/Apps/Artist/Routes/Overview/state.ts [hunk 5]
```diff
@@ -26,7 +30,9 @@ export class FilterState extends Container<State> {
         if (props[filter]) {
           if (filter === "major_periods") {
             this.state[filter] = [props[filter]]
-          } else if (filter === "for_sale") {
+          } else if (
+            ["for_sale", "ecommerce", "at_auction"].indexOf(filter) !== -1
```
[the remark this answers is not in the document]
Comment (mzikherman) [reply src/Apps/Artist/Routes/Overview/state.ts]: ah right!

**Comment by damassi** [comment]
Ah shoot, I merged too soon :/

**Comment by mzikherman** [comment]
I"ll PR the update and a release commit as well, might as well get this in Force.

**Pull request merged**